pyinterprod/uniprot/unirule.py

```diff
@@ -707,3 +707,216 @@ def update_signatures(filepath: str, uri: str):
     con.commit()
     cur.close()
     con.close()
+
+
+MAX_DOM_BY_GROUP = 20
+DOM_OVERLAP_THRESHOLD = 0.3
+REPR_DOM_DATABASES = ["H", "J", "M", "R", "N"]
+
+
+def get_repr_domains(ora_url: str, output: str = "repr_domains.tsv"):
+    con = oracledb.connect(ora_url)
+    cur = con.cursor()
+
+    placeholders = ', '.join(':' + str(i + 1) for i in range(len(REPR_DOM_DATABASES)))
+    params = {str(i + 1): db for i, db in enumerate(REPR_DOM_DATABASES)}
+    cur.execute(
+        f"""
+        SELECT PROTEIN_AC, H.METHOD_AC, H.DBCODE, POS_FROM, POS_TO, FRAGMENTS
+        FROM INTERPRO.MATCH H
+        INNER JOIN INTERPRO.METHOD D
+        ON H.METHOD_AC = D.METHOD_AC
+        WHERE H.DBCODE in ({placeholders})
+        AND (D.SIG_TYPE = 'D' OR D.SIG_TYPE = 'R')
+        ORDER BY PROTEIN_AC
+        """, params
+    )
+   
+
+    logger.info(f"Writing {output}")
+    previous_protein_acc = None
+    domains = []
+    with open(output, "w") as f:
+        for protein_acc, signature_acc, dbcode, pos_start, pos_end, frags_str in cur:
+            frag_list = _get_fragments(pos_start, pos_end, frags_str)
+            match = {
+                "signature": signature_acc,
+                "start": pos_start,
+                "end": pos_end,
+                "frag": frags_str,
+                "fragments": frag_list,
+                "rank": REPR_DOM_DATABASES.index(dbcode)
+            }
+            if previous_protein_acc is None:
+                previous_protein_acc = protein_acc
+            if protein_acc != previous_protein_acc:
+                repr_domains = _select_repr_domains(domains)
+                for domain in repr_domains:
+                    f.write(
+                        f"{previous_protein_acc}\t{domain['signature']}\t{domain['start']}\t{domain['end']}\t{domain['frag']}\n"
+                    )
+                domains = []
+                previous_protein_acc = protein_acc
+
+            domains.append(match)
+
+        if domains:
+            repr_domains = _select_repr_domains(domains)
+            for domain in repr_domains:
+                f.write(
+                    f"{protein_acc}\t{domain['signature']}\t{domain['start']}\t{domain['end']}\t{domain['frag']}\n"
+                )
```

Comment:
PEP 8 (line too long)

```suggestion
                f.write(
                    f"{protein_acc}\t{domain['signature']}\t{domain['start']}\t"
                    f"{domain['end']}\t{domain['frag']}\n"
                )
```